e2e-polybft/e2e/nexus_bridge_test.go

```diff
@@ -145,3 +149,943 @@ func TestE2E_ApexBridge_Nexus(t *testing.T) {
+
+		ethBalanceAfter, err := cardanofw.GetEthAmount(ctx, apex.Nexus, userNexus)
+		require.NoError(t, err)
+		fmt.Printf("ETH Amount after Tx %d\n", ethBalanceAfter)
+	})
+
+	t.Run("From Prime to Nexus one by one - wait for other side", func(t *testing.T) {
+		sendAmount := uint64(1)
+		sendAmountDfm := new(big.Int).SetUint64(sendAmount)
+		exp := new(big.Int).Exp(big.NewInt(10), big.NewInt(6), nil)
+		sendAmountDfm.Mul(sendAmountDfm, exp)
+
+		expectedAmount := ethgo.Ether(sendAmount)
+
+		instances := 5
+
+		for i := 0; i < instances; i++ {
+			ethBalanceBefore, err := cardanofw.GetEthAmount(ctx, apex.Nexus, userNexus)
+			fmt.Printf("ETH Amount before Tx %d\n", ethBalanceBefore)
+			require.NoError(t, err)
+
+			txHash, err := userPrime.BridgeNexusAmount(t, ctx, txProviderPrime, apex.Bridge.PrimeMultisigAddr,
+				receiverAddrNexus, sendAmountDfm.Uint64(), apex.PrimeCluster.NetworkConfig(), receiverAddrNexus)
+			require.NoError(t, err)
+
+			fmt.Printf("Tx sent. hash: %s\n", txHash)
+
+			ethExpectedBalance := big.NewInt(0).Add(ethBalanceBefore, expectedAmount)
+
+			err = cardanofw.WaitForEthAmount(context.Background(), apex.Nexus, userNexus, func(val *big.Int) bool {
+				return val.Cmp(ethExpectedBalance) == 0
+			}, 30, 30*time.Second)
+			require.NoError(t, err)
+		}
+
+		ethBalanceAfter, err := cardanofw.GetEthAmount(ctx, apex.Nexus, userNexus)
+		fmt.Printf("ETH Amount after Tx %d\n", ethBalanceAfter)
+		require.NoError(t, err)
+	})
+
+	t.Run("From Prime to Nexus one by one - don't wait for other side", func(t *testing.T) {
+		sendAmount := uint64(1)
+		sendAmountDfm := new(big.Int).SetUint64(sendAmount)
+		exp := new(big.Int).Exp(big.NewInt(10), big.NewInt(6), nil)
+		sendAmountDfm.Mul(sendAmountDfm, exp)
+
+		expectedAmount := ethgo.Ether(sendAmount)
+
+		ethBalanceBefore, err := cardanofw.GetEthAmount(ctx, apex.Nexus, userNexus)
+		fmt.Printf("ETH Amount before Tx %d\n", ethBalanceBefore)
+		require.NoError(t, err)
+
+		instances := 5
+
+		for i := 0; i < instances; i++ {
+			txHash, err := userPrime.BridgeNexusAmount(t, ctx, txProviderPrime, apex.Bridge.PrimeMultisigAddr,
+				receiverAddrNexus, sendAmountDfm.Uint64(), apex.PrimeCluster.NetworkConfig(), receiverAddrNexus)
+			require.NoError(t, err)
+
+			fmt.Printf("Tx sent. hash: %s\n", txHash)
+		}
+
+		transferedAmount := new(big.Int).SetInt64(int64(instances))
+		transferedAmount.Mul(transferedAmount, expectedAmount)
+		ethExpectedBalance := big.NewInt(0).Add(ethBalanceBefore, transferedAmount)
+
+		err = cardanofw.WaitForEthAmount(context.Background(), apex.Nexus, userNexus, func(val *big.Int) bool {
+			return val.Cmp(ethExpectedBalance) == 0
+		}, 30, 30*time.Second)
+		require.NoError(t, err)
+
+		ethBalanceAfter, err := cardanofw.GetEthAmount(ctx, apex.Nexus, userNexus)
+		fmt.Printf("ETH Amount after Tx %d\n", ethBalanceAfter)
+		require.NoError(t, err)
+	})
+
+	t.Run("From Prime to Nexus parallel", func(t *testing.T) {
+		sendAmount := uint64(1)
```

Comment:
this code seams duplicate. Make inner function which receives amount and returns `(expectedAmount, uint64(sendAmountDFM))`